sa/sa.go

```diff
@@ -1493,3 +1494,158 @@ func (ssa *SQLStorageAuthority) UnpauseAccount(ctx context.Context, req *sapb.Re
 
 	return total, nil
 }
+
+// AddRateLimitOverride adds a rate limit override to the database. Only the
+// comment field is optional. If the override already exists, it will be
+// updated. If the override does not exist, it will be inserted and enabled. If
+// the override exists but has been disabled, it will be updated but not be
+// re-enabled. The status of the override is returned in Enabled field of the
+// response. To re-enable an override, use the EnableRateLimitOverride method.
+func (ssa *SQLStorageAuthority) AddRateLimitOverride(ctx context.Context, req *sapb.AddRateLimitOverrideRequest) (*sapb.AddRateLimitOverrideResponse, error) {
+	if core.IsAnyNilOrZero(req, req.Override, req.Override.LimitEnum, req.Override.BucketKey, req.Override.Count, req.Override.Burst, req.Override.Period) {
+		return nil, errIncompleteRequest
+	}
+
+	var inserted bool
+	var enabled bool
+	now := ssa.clk.Now()
+
+	_, err := db.WithTransaction(ctx, ssa.dbMap, func(tx db.Executor) (any, error) {
+		var existing overrideModel
+		err := tx.SelectOne(ctx, &existing, `
+			SELECT enabled
```

Comment:
This does result in a slightly unfortunate situation where we have a model object floating around that we could never insert into the database. It might be cleaner to do:
```suggestion
		var alreadyEnabled bool
		err := tx.SelectOne(ctx, &alreadyEnabled, `
			SELECT enabled
```